packages/worker/src/index.ts

```diff
@@ -5,60 +5,126 @@ import {
+	TranscriptionOutput,
 } from '@guardian/transcription-service-common';
 import { getSNSClient, publishTranscriptionOutput } from './sns';
+import {
+	getTranscriptionText,
+	convertToWav,
+	createContainer,
+} from './transcribe';
+import {
+	getFile,
+	getS3Client,
+} from '@guardian/transcription-service-common/src/s3';
+import path from 'path';
 
 const main = async () => {
-	const config = await getConfig();
-	const client = getClient(config.aws.region, config.aws.localstackEndpoint);
-
-	// to simulate a transcription job, delay 5 seconds in DEV, 2 minutes in PROD before deleting the message
-	const dummyDelay = config.app.stage === 'DEV' ? 5 : 120;
-	// override timeout to allow enough time for our dummy delay before message becomes available
-	const timeoutOverride = dummyDelay + 1;
-	const message = await getNextMessage(
-		client,
-		config.app.taskQueueUrl,
-		timeoutOverride,
-	);
+	try {
+		const config = await getConfig();
 
-	const snsClient = getSNSClient(
-		config.aws.region,
-		config.aws.localstackEndpoint,
-	);
+		const numberOfThreads = config.app.stage === 'PROD' ? 16 : 2;
 
-	if (isFailure(message)) {
-		return;
-	}
-	console.log(config.app.taskQueueUrl);
-
-	await publishTranscriptionOutput(
-		snsClient,
-		config.app.destinationTopicArns.transcriptionService,
-		{
-			id: 'test-id',
-			transcriptionSrt: 'test-srt',
-			languageCode: 'en',
-			userEmail: 'test@test.com',
-		},
-	);
+		const client = getClient(config.aws.region, config.aws.localstackEndpoint);
+		const message = await getNextMessage(client, config.app.taskQueueUrl);
+		const snsClient = getSNSClient(
+			config.aws.region,
+			config.aws.localstackEndpoint,
+		);
+
+		if (isFailure(message)) {
+			console.error(`Failed to fetch message due to ${message.errorMsg}`);
+			return;
+		}
+
+		if (!message.message) {
+			console.log('No messages available');
+			return;
+		}
 
-	if (message.message?.ReceiptHandle) {
 		const job = parseTranscriptJobMessage(message.message);
+
 		console.log(
 			`Fetched transcription job with id ${message.message.MessageId}}`,
 			job,
 		);
-		// wait for 3 minutes then delete the message to simulate the transcription
-		setTimeout(() => {
+
+		if (!job) {
+			console.error('Failed to parse job message', message);
+			return;
+		}
+
+		const fileToTranscribe = await getFileFromS3(config, job?.s3Key);
+
+		console.log('file is here');
+
+		// docker container to run ffmpeg and whisper on file
+		const containerId = await createContainer(path.parse(fileToTranscribe).dir);
+
+		const ffmpegResult = await convertToWav(containerId, fileToTranscribe);
+
+		if (
+			message.message?.ReceiptHandle &&
+			ffmpegResult.duration &&
+			ffmpegResult.duration !== 0
+		) {
+			// Adding 300 seconds (5 minutes) and the file duration
+			// to allow time to load the whisper model
+			await changeMessageVisibility(
+				client,
+				config.app.taskQueueUrl,
+				message.message.ReceiptHandle,
+				ffmpegResult.duration + 300,
```

Comment:
should we consider adding a higher overhead than 5 minutes?
maybe something like `ffmpegResult.duration * 1.5 + 300` to reduce the risk that a long-running transcription results in another worker picking up the job?

Reply:
I'm not sure what the right figure here is but in theory, if running on our target instance type and if our limited performance testing is accurate for other file types, we should be transcribing files in less than half the length of the file, so I think `duration + 300` might be a good first guess.

Having said that, there's no real reason to abandon a file just because it's taking 2x our expected time so maybe we should set this to a v high figure, I guess it depends what's more likely - a transcription taking longer than expected vs some error happening that causes the transcript to hang for ages at which point it might be better to allocate it to another worker. I suspect probably the first option is more likely! e.g. we haven't tested transcribe time for files in languages other than english

Reply:
ok, I thought our average transcription times were about the same as the length of the files rather than half!
happy to keep the current visibility timeout and we can adjust while testing if necessary

Reply:
Thanks both for giving your ideas. For now I leave to as `duration + 300` but will with a bigger not-english file today and see if it takes any longer